great_expectations/render/renderer_configuration.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass, field
+from typing import Union
+
+from great_expectations.core import (
+    ExpectationConfiguration,
+    ExpectationValidationResult,
+)
+
+
+@dataclass(frozen=True)
+class RendererConfiguration:
+    """Configuration object built for each renderer."""
+
+    configuration: Union[ExpectationConfiguration, None]
+    result: Union[ExpectationValidationResult, None]
```

Comment:
I would prefer for these attributes to be more explicitly named (`expectation_configuration` and `expectation_validation_result`), but this follows the pattern for the existing renderer signatures. In the past it seems we ended up using abbreviations for `expectation_validation_result` (such as `evr` or `result`) in a lot of places to reduce verbosity. Changing these attributes to be more explicitly named would represent a move in that direction everywhere since I am proposing that this class be instantiated in the renderers themselves for now.